### DISCUSSION TITLE
sh: <MY_PACKAGE_BIN_NAME>: command not found when used with npx

### DISCUSSION THREAD

**Discussion by pawelpiotrowski** [jmcdo29/nest-commander#910] [Q&A]
Hello,

I just started with nest-commander but I can't get example taken from the docs running.

Published package to npm (nestcms-cli), here is my package.json:

```
{
  "name": "nestcms-cli",
  "version": "0.0.1",
  "description": "",
  "author": "Pawel Piotrowski (https://github.com/pawelpiotrowski)",
  "license": "MIT",
  "bin": {
    "nestcms-cli": "dist/main"
  },
  "scripts": {
    "build": "nest build",
    "format": "prettier --write \"src/**/*.ts\"",
    "lint": "eslint \"src/**/*.ts\" --fix",
    "start": "nest start",
    "start:dev": "nest start --watch",
    "test": "jest",
    "test:watch": "jest --watch",
    "test:cov": "jest --coverage",
    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
    "version": "node ./scripts/version.js"
  },
  "dependencies": {
    "@nestjs/common": "^9.0.0",
    "@nestjs/core": "^9.0.0",
    "nest-commander": "^3.7.1",
    "reflect-metadata": "^0.1.13"
  },
  "devDependencies": {
    "@nestjs/cli": "^9.0.0",
    "@nestjs/schematics": "^9.0.0",
    "@nestjs/testing": "^9.0.0",
    "@types/jest": "29.5.0",
    "@types/node": "18.15.11",
    "@typescript-eslint/eslint-plugin": "^5.0.0",
    "@typescript-eslint/parser": "^5.0.0",
    "eslint": "^8.0.1",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-prettier": "^4.0.0",
    "jest": "29.5.0",
    "prettier": "^2.3.2",
    "source-map-support": "^0.5.20",
    "ts-jest": "29.0.5",
    "ts-loader": "^9.2.3",
    "ts-node": "^10.0.0",
    "tsconfig-paths": "4.2.0",
    "typescript": "^4.7.4"
  },
  "jest": {
    "moduleFileExtensions": [
      "js",
      "json",
      "ts"
    ],
    "rootDir": "src",
    "testRegex": ".*\\.spec\\.ts$",
    "transform": {
      "^.+\\.(t|j)s$": "ts-jest"
    },
    "collectCoverageFrom": [
      "**/*.(t|j)s"
    ],
    "coverageDirectory": "../coverage",
    "testEnvironment": "node"
  },
  "bugs": {
    "url": "***"
  },
  "homepage": "***"
}

```
But when trying to execute basic command:
`npx nestcms-cli basic -s hello`
I am getting `sh: nestcms-cli: command not found`
When testing locally I can see the output correctly:
`npm run start -- -- basic -s hello`
or
`node dist/main basic -s hello`

-> `{ param: [], string: 'hello' }`

Have I missed something?

Link to repo: https://github.com/pawelpiotrowski/nestcms-cli
Thanks!

**Comment by jmcdo29** [comment]
Did you run the global install for the command to register the `bin` with your machine's global `PATH` variable?

**Reply by pawelpiotrowski** [reply]
Hi @jmcdo29 
Thanks. I have after this comment with the same result: 
`npm install nestcms-cli -g`
`nestcms-cli basic -s hello`
`zsh: command not found: nestcms-cli`

I haven't done that before because if running with npx package doesn't need to be installed at all - does it?
No trouble running angular-cli nest-cli or recently I tested playwright and all of them are working fine with npx and no installation local or global

**Reply by pawelpiotrowski** [reply]
... and just in case if you haven't time to check content of the repo I posted I am using example from [Putting all together example from README](https://github.com/jmcdo29/nest-commander#putting-it-all-together) but without LoggerService dependency directly calling `console.log`.
